web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte

```diff
@@ -1,41 +1,163 @@
 <script lang="ts">
-  import { goto } from "$app/navigation";
+  import {
+    afterNavigate,
+    beforeNavigate,
+    goto,
+    replaceState,
+  } from "$app/navigation";
   import { page } from "$app/stores";
   import ErrorPage from "@rilldata/web-common/components/ErrorPage.svelte";
   import { getStateManagers } from "@rilldata/web-common/features/dashboards/state-managers/state-managers";
   import { metricsExplorerStore } from "@rilldata/web-common/features/dashboards/stores/dashboard-stores";
   import type { MetricsExplorerEntity } from "@rilldata/web-common/features/dashboards/stores/metrics-explorer-entity";
   import { convertExploreStateToURLSearchParams } from "@rilldata/web-common/features/dashboards/url-state/convertExploreStateToURLSearchParams";
+  import { getExploreStateFromSessionStorage } from "@rilldata/web-common/features/dashboards/url-state/getExploreStateFromSessionStorage";
+  import { getUpdatedUrlForExploreState } from "@rilldata/web-common/features/dashboards/url-state/getUpdatedUrlForExploreState";
   import {
     createQueryServiceMetricsViewSchema,
     type V1ExplorePreset,
   } from "@rilldata/web-common/runtime-client";
   import type { HTTPError } from "@rilldata/web-common/runtime-client/fetchWrapper";
   import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import type { AfterNavigate } from "@sveltejs/kit";
+  import { onMount } from "svelte";
 
+  export let metricsViewName: string;
+  export let exploreName: string;
   export let defaultExplorePreset: V1ExplorePreset;
-  export let partialExploreState: Partial<MetricsExplorerEntity>;
+  export let exploreStateFromYAMLConfig: Partial<MetricsExplorerEntity>;
+  export let initExploreState: Partial<MetricsExplorerEntity> | undefined =
```

Comment:
"initExploreState" doesn't specify which source of state it is.

Reply:
Hmmm this is supposed to be bookmark state on cloud explore, token's state for public URLs and undefined for rill-dev. Not sure what a good name would be to capture this. Adding them explicitly like `bookmarkExploreState` etc doesnt make sense since it is irrelevant on rill-dev.

Reply:
I think optional props would be most clear: 
```
export let bookmarkExploreState: Partial<ExploreState> | undefined = undefined;
export let tokenExploreState: Partial<ExploreState> | undefined = undefined;
```

Reply:
I have renamed the variables until the call to DashboardURLStateSync. We should avoid having cloud/dev specific login in common code.